### ISSUE TITLE
Contact deletion management

### ISSUE THREAD

**Issue by skjnldsv** [nextcloud/contacts#296]
Two options (cumulable):
 - A trashbin to restore deleted contacts (local save)
 - An undo button

**Comment by frogueat** [comment]
Nice would also be the posibility, to delete many contacts at once. Eg when imported something via vcf and the data is wrong ... Maybe a whole group?

**Comment by skjnldsv** [comment]
Not the same issue, see #277

**Comment by MariusBluem** [comment]
In the same way as in the calendar app 🙈

**Comment by fgarcia78** [comment]
When there is more than one addressbook and the user deletes one of them (by clicking the trash button), the UI _doesn't confirm that action_ and all contacts in such addressbook get wiped; furthermore, the UI does not visually reflect the deletion and keeps showing those contacts in the list (cache?).

I don't know if this is addressed elsewhere, I'm in 1.5.3.

**Comment by skjnldsv** [comment]
@fgarcia78 This has literaly nothing to do with that issue! 😕 
Lucky for you, this issue had already been fixed and will be on the next release. See #288 .

**Comment by fgarcia78** [comment]
@skjnldsv - yeah, I see now my error... altough the ticket name is misleading: "Contact deletion management" sounds too broad and this enhancement refers to "Contact _un-deletion_ management"!! 😄

**Comment by Jas2Ma** [comment]
Add nextcloud activity events for contacts, so deletion of a contact can be recognized by other users of the address book.

**Comment by tdoerschel** [comment]
I agree that the risk of accidentelly deleting a contact or a field is very high: The delete buttons are all over the webpage.

A confirmation popup/diaglog (are you sure)?, trashbin or undo (for 3 seconds) would ease one's mind.

**Comment by tdoerschel** [comment]
Feature has been implemented for contact delete in version 3.1.0

Can you also also do it for the field delete buttons?

**Comment by PhilippSchlesinger** [comment]
@mejo- could you maybe give a hint (or even some screenshot) how this feature got implemented in the meantime? Did I really miss that out? When I delete a contact, I is simply gone. There is no undo dialog and no trashbin to restore from.
(Contacts 5.2.0 on Nextcloud 25.0.6)

Thank you.

**Comment by mejo-** [comment]
Sorry, I got confused. You're right and there's no delete undo implemented yet. Sorry for the noise :see_no_evil:

**Comment by joshtrichards** [comment]
> Sorry, I got confused. You're right and there's no delete undo implemented yet. Sorry for the noise 🙈

To be fair, there used to be. #1025
